### ISSUE TITLE
AI is beatable

### ISSUE THREAD

**Issue by tehp** [ttsiodras/Score4#4]
Hi Thanassis. 

I found your project very interesting!

I noticed is that your AI is beatable. Was your AI intended to be beatable? 

Have you read this related paper? 
http://www.informatik.uni-trier.de/~fernau/DSL0607/Masterthesis-Viergewinnt.pdf

**Comment by ttsiodras** [comment]
> I found your project very interesting! ...

Thanks. The single point of this project (as you can read in the [blog post](https://www.thanassis.space/score4.html) I wrote about it 6 years ago, was to "play" with an AI algorithm implemented in functional-style. More specifically, my first implementation was in OCaml - which is analyzed in detail in the post. 

I then wanted to see how F# - arguably, a "daughter" of OCaml - fares in terms of speed... So it then became a benchmarking game, with my C/C++ version eventually dominating all others. I also coded it in LISP and used macros to do loop-unrolling at compile-time... which made my Common LISP version 2nd fastest amongst all other languages (see the repo for details).

I then wanted to see the algorithm in a browser - so I coded it in Javascript...  which showed that Firefox was much slower than Chrome in executing the code, so I [raised a bug report](https://bugzilla.mozilla.org/show_bug.cgi?id=675167)...   and soon afterwards, as you can see in the comments of that bug report, Firefox engineers fixed the issue and made Firefox just as fast as Chrome :-)

In plain words, this was just a toy - the kind nerds play with :-)

Hope you enjoyed using it!

**Comment by tehp** [comment]
Ah yes, this is what I figured. 

Thanks for the cool project!
